featurebyte/models/request_input.py

```diff
@@ -148,9 +157,44 @@ async def materialize(
                 columns = self.columns
             query_expr = select_and_rename_columns(query_expr, columns, self.columns_rename_mapping)
 
+        # apply time range filter if sample_from_timestamp and sample_to_timestamp are provided
+        time_range_conditions: list[expressions.Expression] = []
+        adapter = get_sql_adapter(session.get_source_info())
+        if sample_from_timestamp is not None:
+            time_range_conditions.append(
+                expressions.GTE(
+                    this=adapter.normalize_timestamp_before_comparison(
+                        quoted_identifier(SpecialColumnName.POINT_IN_TIME)
+                    ),
+                    expression=make_literal_value(
+                        sample_from_timestamp.isoformat(), cast_as_timestamp=True
+                    ),
+                )
+            )
+        if sample_to_timestamp is not None:
+            time_range_conditions.append(
+                expressions.LT(
+                    this=adapter.normalize_timestamp_before_comparison(
+                        quoted_identifier(SpecialColumnName.POINT_IN_TIME)
+                    ),
+                    expression=make_literal_value(
+                        sample_to_timestamp.isoformat(), cast_as_timestamp=True
+                    ),
+                )
+            )
+        if time_range_conditions:
+            if self.columns is None and self.columns_rename_mapping is None:
+                # add a select * in case the expression already has a where clause
+                query_expr = expressions.Select(expressions=[expressions.Star()]).from_(
+                    query_expr.subquery()
+                )
+            if len(time_range_conditions) == 1:
```

Comment:
Is this condition necessary? My impression is that `expressions.And` should work even if `time_range_conditions` has a single item.

Reply:
Looks like it's not. Will remove this